<a href="https://colab.research.google.com/github/YomnaImad07/FlyRank-ML-Internship/blob/main/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/YomnaImad07/FlyRank-ML-Internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.

**Lane: Ranking Signal Analysis**

I'm choosing the Ranking Signal Analysis lane. This lane lets me study which observed content and search signals (age, word count, average position, freshness) are associated with visibility, clicks, and engagement — without needing to define a complex future-window target yet. It's a strong first step before moving into a heavier prediction or scoring lane in later weeks.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


**Decision:** Which content signals should an SEO content editor prioritize when reviewing pages for improvement?

**Who acts:** A content/SEO editor or analyst reviewing the content inventory.

**Cost of a wrong call:** If we flag a weak signal as important, the editor wastes limited review time on pages unlikely to improve. If we miss a real signal, the editor overlooks pages that could have been fixed for real gains — a lost opportunity cost.

**Why data/ML helps:** The relationship between content signals (age, freshness, word count, position, CTR) and performance is not obvious from a single rule — multiple signals interact and shift over time across 32 clients. A simple if-statement can't capture that; grouped analysis and correlations can.

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [ ]:
!git clone https://github.com/YomnaImad07/FlyRank-ML-Internship

fatal: destination path 'FlyRank-ML-Internship' already exists and is not an empty directory.


In [ ]:
import pandas as pd

df = pd.read_csv("/content/FlyRank-ML-Internship/data/raw/content_refresh_anonymized.csv")

print("Total rows:", len(df))
print("Unique clients:", df["client_id"].nunique())
print("Rows with avg_position == 0 (no data):", (df["avg_position"] == 0).sum())
print("Share of pages with trend_direction == 'down':", (df["trend_direction"] == "down").mean())
print("Mean impressions_90d:", df["impressions_90d"].mean())

Total rows: 30000
Unique clients: 32
Rows with avg_position == 0 (no data): 1205
Share of pages with trend_direction == 'down': 0.5420666666666667
Mean impressions_90d: 5200.3663


The starter dataset contains 30,000 rows across 32 unique clients — enough scale and enough client diversity to look for signals that generalize, not just quirks of one client. About 54.2% of pages show a declining trend (trend_direction == "down"), which gives a well-balanced set of both declining and non-declining pages to compare and contrast — not a rare-event problem. Average 90-day impressions is about 5,200, showing most pages in this sample carry real search visibility worth analyzing. Note that 1,205 rows have avg_position == 0, which per the data dictionary means "no position data" rather than rank zero — these rows will need to be filtered or flagged separately, not treated as a top ranking.

These numbers suggest this lane is worth the next 7 weeks: there's enough volume, enough client diversity, and a roughly even split between declining and stable pages to find real, generalizable signal patterns rather than noise.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


**What I can claim:** Observed associations between content signals and performance metrics (e.g., "pages with lower word count are observed to have lower average impressions"). Directional patterns ("higher content age is directionally associated with declining trend"). Decision-support guidance for editors, not automated actions.

**What I will never claim:** Causal proof (I will not say a signal *causes* a performance change without a controlled experiment). I will not claim to have discovered a Google ranking algorithm factor. I will not treat `trend_direction` or `trend_pct` as features — they define the label itself, not a predictor, per the flyrank-data skill's label trap warning.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.